pkg/webservice/state_dump.go

```diff
@@ -0,0 +1,207 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package webservice
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/apache/incubator-yunikorn-core/pkg/log"
+	metrics2 "github.com/apache/incubator-yunikorn-core/pkg/metrics"
+	"github.com/apache/incubator-yunikorn-core/pkg/scheduler"
+	"github.com/apache/incubator-yunikorn-core/pkg/webservice/dao"
+
+	"github.com/gorilla/mux"
+	"go.uber.org/zap"
+)
+
+const (
+	defaultStateDumpPeriod = 60
```

Comment:
From this line we cannot know what this 60 is? mins? seconds? Since this is a duration, please use Duration type instead of int, or if it is not possible please rename the variable to mirror the exact value.

Reply:
Creating the state dump by default every minute seems a little too aggressive for me. It will make huge files. I think we should print the dumps less freqvently. What is the default value in yarn for the state dump creeation?

Reply:
> Creating the state dump by default every minute seems a little too aggressive for me. It will make huge files. I think we should print the dumps less freqvently. What is the default value in yarn for the state dump creeation?

Fair Scheduler performs this step in every 25 `update()` call (update is called for every heartbeat). To me, a state dump per minute does not sound that aggressive.